serverless/functions/join-stream-as-speaker.js

```diff
@@ -48,156 +47,80 @@ module.exports.handler = async (context, event, callback) => {
-      .syncMapPermissions(user_identity)
-      .update({ read: true, write: false, manage: false })
-  } catch (e) {
-    response.setStatusCode(500);
-    response.setBody({
-      error: {
-        message: 'error adding read access to speakers map',
-        explanation: e.message,
-      },
-    });
-    return callback(null, response);
-  }
-  
-  const raisedHandsMapName = `raised_hands`;
+  await streamSyncClient
+    .syncMaps('speakers')
+    .syncMapPermissions(user_identity)
+    .update({ read: true, write: false, manage: false })
+    .catch(createErrorHandler('error adding read access to speakers map'));
+
   // Give user read access to raised hands map
-  try {
-    await streamSyncClient.syncMaps(raisedHandsMapName)
-      .syncMapPermissions(user_identity)
-      .update({ read: true, write: false, manage: false });
-  } catch (e) {
-    response.setStatusCode(500);
-    response.setBody({
-      error: {
-        message: 'error adding read access to raised hands map',
-        explanation: e.message,
-      },
-    });
-    return callback(null, response);
-  }
+  await streamSyncClient
+    .syncMaps('raised_hands')
+    .syncMapPermissions(user_identity)
+    .update({ read: true, write: false, manage: false })
+    .catch(createErrorHandler('error adding read access to raised hands map'));
 
   // Give user read access to viewers map
-  try {
-    await streamSyncClient.syncMaps('viewers')
-      .syncMapPermissions(user_identity)
-      .update({ read: true, write: false, manage: false })
-  } catch (e) {
-    response.setStatusCode(500);
-    response.setBody({
-      error: {
-        message: 'error adding read access to viewers map',
-        explanation: e.message,
-      },
-    });
-    return callback(null, response);
-  }
+  await streamSyncClient
+    .syncMaps('viewers')
+    .syncMapPermissions(user_identity)
+    .update({ read: true, write: false, manage: false })
+    .catch(createErrorHandler('error adding read access to viewers map'));
 
   const conversationsClient = client.conversations.services(CONVERSATIONS_SERVICE_SID);
 
-  try {
-    // Find conversation
-    conversation = await conversationsClient.conversations(room.sid).fetch();
-  } catch (e) {
-    console.error(e);
-    response.setStatusCode(500);
-    response.setBody({
-      error: {
-        message: 'error finding conversation',
-        explanation: 'Something went wrong when finding a conversation.',
-      },
+  // Find conversation
+  conversation = await conversationsClient
+    .conversations(room.sid)
+    .fetch()
+    .catch(createErrorHandler('error finding conversation'));
+
+  // Add participant to conversation
+  await conversationsClient
+    .conversations(room.sid)
+    .participants.create({ identity: user_identity })
+    .catch((error) => {
+      // Ignore "Participant already exists" error (50433)
+      if (error.code !== 50433) {
+        createErrorHandler('error adding read access to user document')(error);
```

Comment:
Should this error message explain that there was an error adding the participant to the conversation?  i.e. `'error creating conversation participant'`